pkg/ipblock/ipblock.go

```diff
@@ -119,9 +121,9 @@ func (b *IPBlock) ipCount() int {
 func (b *IPBlock) Split() []*IPBlock {
 	res := make([]*IPBlock, len(b.ipRange.IntervalSet))
 	for index, ipr := range b.ipRange.IntervalSet {
-		newBlock := IPBlock{}
-		newBlock.ipRange.IntervalSet = append(newBlock.ipRange.IntervalSet, interval.Interval{Start: ipr.Start, End: ipr.End})
-		res[index] = &newBlock
+		res[index] = &IPBlock{
+			ipRange: interval.FromInterval(ipr.Start, ipr.End),
```

Comment:
Maybe we should change `interval.FromInterval` to take an `Interval` object and have another function `FromRange` that takes `start` and `end` parameters.

Reply:
I agree (or maybe add a `ToSet()` method). I don't think it should be part of this PR though.